### PR TITLE
fix: add validation error on NFT import

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**
Adds an error msg if contract address is not validation on the NFT import modal.

(Just to make the behavior more consistent with the token-import modal)

## **Manual testing steps**

_1. Click on "import nft" button in the NFT tab
_2. Add invalid contract address and notice error msg

### **Before**

image

### **After**

image

### **Fixes**
https://consensyssoftware.atlassian.net/browse/MMASSETS-35

## **Pre-merge author checklist**

- [x ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [x] What problem this PR is solving.
  - [ x] How this problem was solved.
  - [ x] How reviewers can test my changes.
- [ x] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [x ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1,2 @@
+nodejs 18.14.0
+yarn 1.22.19
```
Comment (salimtb) [inline .tool-versions:2]: there's no need to include the .tool-version , you can ignore it

File: ui/components/multichain/import-nfts-modal/import-nfts-modal.js [hunk 2]
```diff
@@ -129,6 +131,10 @@ export const ImportNftsModal = ({ onClose }) => {
   };
 
   const validateAndSetAddress = (val) => {
+    setNftAddressValidationError(null);
+    if (!isValidHexAddress(val)) {
+      setNftAddressValidationError(t('invalidAddress'));
+    }
```
Comment (salimtb) [inline ui/components/multichain/import-nfts-modal/import-nfts-modal.js:137]: is there any unit test for this function ? if so , you can add the if case and test it

File: .tool-versions (hunk 1 above)
Comment (sahar-fehri) [reply .tool-versions:2]: that one slipped, deleting

File: ui/components/multichain/import-nfts-modal/import-nfts-modal.js (hunk 2 above)
Comment (sahar-fehri) [reply ui/components/multichain/import-nfts-modal/import-nfts-modal.js:137]: pushed! 🙏

**Review by salimtb — DISMISSED** [review]

**Review by NidhiKJha — COMMENTED** [review]
This looks really good @sahar-fehri , before I approve one follow-up question. Do we want to show the error message when the address field is blank, I think in this case we should not show any message.
Screenshot 2023-09-26 at 12 03 04 PM

**Review by NidhiKJha — APPROVED** [review]
LGTM!

**Review by salimtb — APPROVED** [review]

**Pull request merged**